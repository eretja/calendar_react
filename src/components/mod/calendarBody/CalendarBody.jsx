import React from "react";
import PropTypes from "prop-types";
import styles from "./calendarBody.scss";
import Button from "ui/buttons/Button";

const CalendarBody = (props) => {
 let {year,month} = props.state
 
 let daysInMonth = new Date(year, month+1, 0).getDate();

  let firstDayOfWeek = new Date(year, month).getDay();

  let lastDay = new Date(year, month, 0).getDate();
  let day = 1;
  let dayOfNextMonth = 0;
  let cellText;
  const tds = (i, currentDate) => {
    let td = [];
    const baseDayClass = "basis class";
    for (let y = 0; y < 7; y++) {
      let dayClass = baseDayClass;
      if (i === 0 && y < firstDayOfWeek - 1) {
        cellText = lastDay - firstDayOfWeek + 2 + y;
        dayClass += " " + styles.calendarNextMonth;
      } else if (day >= daysInMonth) {
        dayOfNextMonth++;
        dayClass += " " + styles.calendarNextMonth;
        cellText = day;
        day = dayOfNextMonth;
      } else {
        cellText = day;
        if (
          day === currentDate.getDate() &&
          year === currentDate.getFullYear() &&
          month === currentDate.getMonth()
        ) {
          dayClass += " " + styles.actualDate;
        }
        day++;
      }
      td.push(
        <td key={y} className={dayClass}>
          {cellText}
        </td>
      );
    }
    return td;
  };
  const trs = currentDate => {
    let tr = [];
    for (let i = 0; i < 5; i++) {
      tr.push(<tr key={i}>{tds(i, currentDate)}</tr>);
    }
    return tr;
  }; 
  
 return (
    <div className="calendarBody">
        <table>
          <tbody>{trs(props.currentDate)}</tbody>
        </table>
      
      {/* <Button text='Add an Event'/>  */}
    </div>
  );
};
//}

CalendarBody.propTypes = {
  /** Description of prop "text". */

  text: PropTypes.string
};

CalendarBody.defaultProps = {
  text: ""
};

export default CalendarBody;
