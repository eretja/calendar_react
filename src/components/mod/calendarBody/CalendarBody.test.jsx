import React from "react";
import CalendarBody from "./CalendarBody";

import { configure } from "enzyme";
const Adapter = require("enzyme-adapter-react-16");
configure({ adapter: new Adapter() });

import { shallow, mount } from "enzyme";

import TestRenderer from "react-test-renderer";

// const setup = (props={}, state=null)=>{
//     return shallow(<CalendarBody {...props}/>)
// }

describe("<CalendarBody />", () => {
  // https://jestjs.io/docs/en/snapshot-testing
  // https://github.com/facebook/jest/issues/2234

  const fixDate = "2019-04-03T15:31:44.892Z";
  // const fixDate = '2019-04-04T15:31:44.00'; // check error

  const RealDate = Date.now;
  global.Date.now = jest.fn(() => new RealDate(fixDate));
  Object.assign(Date, RealDate);

  //console.log( 'new Date()', new Date(), typeof new Date() );

  let CalendarBodyComponent;
  let state = {
    currentDate: new Date()
  };
  CalendarBodyComponent = shallow(
    <CalendarBody currentDate={state.currentDate} />
  );

  const testRenderer = TestRenderer.create(
    <CalendarBody currentDate={state.currentDate} />
  ).toJSON();

  it("should render CalendarBody with correct html", () => {
    expect(testRenderer).toMatchSnapshot();
  });

  global.Date.now = RealDate;

  it("should mount CalendarBody sucessfully", () => {
    expect(CalendarBodyComponent.length).toBe(1);
  });
});
