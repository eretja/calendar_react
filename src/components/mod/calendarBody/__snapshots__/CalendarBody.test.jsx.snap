// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CalendarBody /> should render CalendarBody with correct html 1`] = `
<div
  className="calendarBody"
>
  <div>
    <table>
      <tbody>
        <tr>
          <td
            className="basis class"
          >
            1
          </td>
          <td
            className="basis class"
          >
            2
          </td>
          <td
            className="basis class"
          >
            3
          </td>
          <td
            className="basis class"
          >
            4
          </td>
          <td
            className="basis class"
          >
            5
          </td>
          <td
            className="basis class"
          >
            6
          </td>
          <td
            className="basis class"
          >
            7
          </td>
        </tr>
        <tr>
          <td
            className="basis class"
          >
            8
          </td>
          <td
            className="basis class"
          >
            9
          </td>
          <td
            className="basis class"
          >
            10
          </td>
          <td
            className="basis class"
          >
            11
          </td>
          <td
            className="basis class"
          >
            12
          </td>
          <td
            className="basis class"
          >
            13
          </td>
          <td
            className="basis class"
          >
            14
          </td>
        </tr>
        <tr>
          <td
            className="basis class"
          >
            15
          </td>
          <td
            className="basis class"
          >
            16
          </td>
          <td
            className="basis class"
          >
            17
          </td>
          <td
            className="basis class"
          >
            18
          </td>
          <td
            className="basis class"
          >
            19
          </td>
          <td
            className="basis class"
          >
            20
          </td>
          <td
            className="basis class"
          >
            21
          </td>
        </tr>
        <tr>
          <td
            className="basis class"
          >
            22
          </td>
          <td
            className="basis class actualDate"
          >
            23
          </td>
          <td
            className="basis class"
          >
            24
          </td>
          <td
            className="basis class"
          >
            25
          </td>
          <td
            className="basis class"
          >
            26
          </td>
          <td
            className="basis class"
          >
            27
          </td>
          <td
            className="basis class"
          >
            28
          </td>
        </tr>
        <tr>
          <td
            className="basis class"
          >
            29
          </td>
          <td
            className="basis class calendarNextMonth"
          >
            30
          </td>
          <td
            className="basis class"
          >
            1
          </td>
          <td
            className="basis class"
          >
            2
          </td>
          <td
            className="basis class"
          >
            3
          </td>
          <td
            className="basis class"
          >
            4
          </td>
          <td
            className="basis class"
          >
            5
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
